### PR TITLE
Update __init__.py

### PR DESCRIPTION
**Pull request by madgrizzle**
This change properly handles chunk sizes that are not exactly 1280 bytes.

### CODE REVIEW

File: ovos_ww_plugin_openwakeword/__init__.py [hunk 1]
```diff
@@ -48,12 +48,12 @@ def update(self, chunk):
         audio_frame = np.frombuffer(chunk, dtype=np.int16).tolist()
         self.audio_buffer.extend(audio_frame)  # build up the buffer until it has enough samples
 
-        if len(self.audio_buffer) >= 1280:
-            # Get prediction from openWakeWord
-            prediction = self.model.predict(self.audio_buffer)
+        while len(self.audio_buffer) >= 1280:
```
Comment (ChanceNCounter) [inline ovos_ww_plugin_openwakeword/__init__.py:52]: Good solution, but needs a stop condition (or a verifiable maximum buffer size that can’t be missed in a refactor)

**Review by ChanceNCounter — CHANGES_REQUESTED** [review]

File: ovos_ww_plugin_openwakeword/__init__.py [hunk 2]
```diff
@@ -48,12 +48,12 @@ def update(self, chunk):
...
-            # Get prediction from openWakeWord
-            prediction = self.model.predict(self.audio_buffer)
+        while len(self.audio_buffer) >= 1280:
+            # Get prediction from openWakeWord.
+            prediction = self.model.predict(self.audio_buffer[:1280])
 
-            # Clear the buffer after each prediction
-            self.audio_buffer = []
+            # Clear the first 1280 bytes from buffer after each prediction
+            del self.audio_buffer[:1280]
```
Comment (ChanceNCounter) [inline ovos_ww_plugin_openwakeword/__init__.py:57]: I’m not sure I knew `del` would operate sanely on a slice, but I’ve just verified that it does. Learn something new after 10 years. Thanks! 🥳

**Comment by goldyfruit** [comment]
`REAME.md` should be updated as well.

File: ovos_ww_plugin_openwakeword/__init__.py (hunk 1 above)
Comment (madgrizzle) [reply ovos_ww_plugin_openwakeword/__init__.py:52]: In practice, I think anytime you are dropping audio you are losing performance.  @JarbasAl mentioned using a cyclical buffer but if that's implemented, there should be info logged that there is a performance issue and the plugin may not be suitable for the platform.

Comment (madgrizzle) [reply ovos_ww_plugin_openwakeword/__init__.py:52]: Also, to enable speex noise suppression that's supported by openWakeWord, you cannot pass a list to the model, but rather an array.  I used 
`prediction = self.model.predict(np.asarray(self.audio_buffer[:1280], dtype=np.int16))` 
but I'm not sure that's efficient.

**Comment by madgrizzle** [comment]
openWakeWord will split chunks into 1280-sample segments so there is no need to split chunks in the plugin.  However, if the chunk size is not a multiple of 1280-samples, then openWakeWord will drop the extra data which will lead to a loss of performance.  A better solution is to default to a multiple of 1280 (perhaps 3840 as that's close to the current 4096) and add to the README that chunk size, if it is changed, should be a multiple of 1280.  We could also detect the chunk size in the plugin initialization portion and raise a warning if its not a multiple.

**Pull request closed without merging**